src/client/responses/DisplayResultsContainer.tsx

```diff
@@ -110,6 +156,16 @@ export const DisplayResultsContainer = (props: Props) => {
     ) {
         return <Loader className="pt-4 pb-4" />;
     }
+    if (
```

Comment:
All these should fall under a `useMemo` instead of writing it explicitly 4 times